docker/docker_image_src.go

```diff
@@ -39,6 +39,12 @@ func newImageSource(ctx *types.SystemContext, ref dockerReference, requestedMani
 	if requestedManifestMIMETypes == nil {
 		requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
 	}
+	for _, mtrequested := range requestedManifestMIMETypes {
+		if !supportedManifestMIMETypesMap()[mtrequested] {
```

Comment:
This is still O(N^2): the call `supportedManifestMIMETypesMap` outside the loop, save it in a variable. Then it is O(N).

(Or don’t bother and just write two nested loops. The arrays are small enough.)